docs/DID Communications Research and Specification/Interactions and Messages.md

```diff
@@ -294,22 +361,392 @@ The <u>authenticator</u> answers with an `authenticationResponse`, providing a `
+
+---
+### credential-revocation
+
+Notifying a holder that a previously issued credential has been revoked. Note that this revocation is declaratory, not constitutive, so the actual revocation has to be done elsewhere (e.g. in the backend of the issuer).
+
+#### Roles
+- <u>**Issuer**</u>: Agent who revokes the credential
+- <u>**Holder**</u>: Agent who holds the credential to be revoked
+
+#### Messages
+
+#### revocation
+The <u>issuer</u> sends the `revocation` to the <u>holder</u>, notifying him of the revocation. The most important field here is `credentialId`, which specifies the credential that has been revoked.
+
+###### Layout
+
+```JSON
+revocation: {
+    "context",
+    "thread",
+    "credentialId",
+    "callbackURL", // OPTIONAL!
+    "responseRequested", //OPTIONAL!
+    "id", // OPTIONAL!
+    "comment", // OPTIONAL!
+    "timing" // OPTIONAL!
+}
+```
+
+
+###### Example(s)
+
+```JSON
+{
+    "context": "credential-revocation/1.0/revocation",
+    "thread": "f7771b285a971ba25d66dbe2d82f0bf5f956f4fe548bdf8617c3f24ebc10ed8c",
+    "credentialId": "credential-69420-delicious-lasagna",
+    "callbackURL": "https://www.aliceswonderland.com/",
+    "id": "did:iota:42edacef81828010b314b96c0915780f206341e0ce8892a1b56678c174eef242",
+    "comment": "Revoked because your Lasagna isn't actually that good."
+}
+```
+
+[Source 1: Aries Revocation Notification](https://github.com/hyperledger/aries-rfcs/tree/master/features/0183-revocation-notification);
+
+---
+### presentation-verification
+
+Proving a set of statements about an identifier.
+
+The credential verification flow is a simple request-response message exchange between the <u>verifier</u> and the <u>prover</u>. The interaction can consist of up to two messages, the first one is OPTIONAL.
+
+#### Roles
+- **Verifier**: Agent who requests a set of Verifiable Credentials with associated requirements
+- **Prover**: Agent who provides a set of Verifiable Credentials in the form of a Verifiable Presentation attempting to satisfy the request
+
+#### Messages
+
+#### presentationRequest
+The <u>verifier</u> requests a set of Verifiable Credentials from the <u>prover</u>. This message is OPTIONAL within this interaction.
+
+###### Layout
+
+```JSON
+presentationRequest: {
+    "context",
+    "thread",
+    "callbackURL",
+    "credentialTypes", // OPTIONAL!
+    "supportedIssuers", // OPTIONAL!
+    "responseRequested", //OPTIONAL!
+    "id", // OPTIONAL!
+    "timing" // OPTIONAL!
+}
+```
+
+###### Example(s)
+
+```JSON
+{
+    "context": "presentation-verification/1.0/presentationRequest",
+    "thread": "f7771b285a971ba25d66dbe2d82f0bf5f956f4fe548bdf8617c3f24ebc10ed8c",
+    "callbackURL": "https://www.bobsworld.com/",
+    "credentialTypes": [
```

Comment:
It would have to be supported Issuers per credentialType

```
"trustedIssuer": [{
     "credentialType" :"YourCatHasAnAttitudeCredential",
     "supportedIssuers": [ 
         "did:iota:58c35471071b3dbb97585ee06bb1dd0239ca338629534296cfbb2db6bc857e21",
         "did:iota:23f0b94812c402a1dea1c424303b178d01485a5dcf26cf977333f3b629bd90ec"
     ]}]
```

Reply:
How do we do that best without introducing yet another field "trustedIssuer" ? -> Discussion Wednesday maybe?

Reply:
followed your suggestion in 9916becd